tests/repository/test_container_image_repository.py

```diff
@@ -31,3 +31,22 @@ def test_registry_id_from_distributor():
     )
 
     assert repo.registry_id == "some/repo"
+
+
+def test_default_registry_id_from_distributor():
+    """default registry_id is used when it's not defined in distributor"""
+    repo = Repository.from_data(
```

Comment:
Shouldn't this be ContainerImageRepository? That's where the logic is implemented.

Reply:
This is fine, as Repository.from_data figures out the concrete subclass to use based on the data.